docs/architecture/adr-102-rpc-companion.md

```diff
@@ -0,0 +1,341 @@
+# ADR-102: RPC Companion
+
+## Changelog
+
+- 2022-03-27: First draft (@andynog)
+
+## Status
+
+Accepted | Rejected | Deprecated | Superseded by
+
+## Context
+
+This ADR proposes an architecture of an ***RPC Companion*** solution implemented based on the proposed [ADR-101 Data Companion Pull API](https://github.com/cometbft/cometbft/blob/thane/adr-084-data-companion-pull-api/docs/architecture/adr-101-data-companion-pull-api.md). This solution can run as a sidecar concurrently with the full node, and it is optional.
+
+This ADR provides a reference implementation of a system that can be used to offload queryable data from a CometBFT
+full node to a database and offering a service that exposes the same JSONRPC methods on a endpoint as the regular JSONRPC methods of a CometBFT node endpoint,
+which makes it easier for integrators of RPC clients such as client libraries and applications to switch to this
+***RPC Companion*** with as minimum effort as possible.
+
+This architecture also makes it possible to scale horizontally the querying capacity of a full node by running multiple
```

Comment:
If this is an important use case, maybe we should revisit the decision we made in ADR101 about the data companion interface only supporting one client. If horizontal scaling becomes the norm, the fact that all RPC-companion clients have to coordinate to "act as one single data-companion client" might entail a lot of overhead.

Reply:
Agreed. Probably we could include this somewhere in this ADR's text

Reply:
added additional text for clarification